### PR TITLE
Added support for star-expanding tuples and numeric tuple indexing

### LINKED ISSUE #6910: sql: support star-expanding label-less tuples + numeric tuple indexing
PR: https://github.com/cockroachdb/cockroach/pull/45609

From release notes:
> CockroachDB now supports accessing the Nth column in a column with tuple type using the syntax `(...).@N`, for example: `SELECT (t).@2 FROM (SELECT (1,'b',2.3) AS t)`. This is a CockroachDB-specific extension. [#45609][#45609] {% comment %}doc{% endcomment %}
>
> CockroachDB now supports expanding all columns of a tuple using the `.*` notation, for example: `SELECT (t).* FROM (SELECT (1,'b',2.3) AS t)`. This is a CockroachDB-specific extension. [#45609][#45609] {% comment %}doc{% endcomment %}

### PR DESCRIPTION
**Pull request by ericharmeling**
Fixes #6910.

- Added description and examples of new tuple selection extensions to Scalar Expressions page.
- Updated output from examples on Scalar Expressions page.

### CODE REVIEW

File: v20.1/scalar-expressions.md [hunk 1]
```diff
@@ -727,16 +713,42 @@ For example:
...
+  ?column?
+------------
+  b
+(1 row)
+~~~
+
+CockroachDB also supports expanding all elements of a tuple as a single row with the `.*` notation. For example:
+{% include copy-clipboard.html %}
+~~~ sql
+> SELECT (t).* FROM (SELECT (1,'b',2.3) AS t)
```
Comment (knz) [inline v20.1/scalar-expressions.md:742]: maybe use an example where the values in the table are named, that will show how the column names are derived from the tuple labels

**Comment by jseldess** [comment]
@ericharmeling, @knz, what's left to do on this PR?

**Comment by ericharmeling** [comment]
@lnhsingh Ping on review :)

**Comment by ericharmeling** [comment]
@lnhsingh No worries! TFTR

**Pull request merged**